### PR TITLE
[ADD] allow_filter on Odoo fields

### PR DESCRIPTION
**Pull request by nhatnm0612**
## What is PR does?

- Add a new field on `ir.model.fieds` model: `allow_filter` - Boolean, default=True.
- Add new abstract model `custom.filter.mixin` to override `fields_get` method: editing searchable attribute on Odoo fields, so if allow_filter is declared as False, field won't be shown on Custom Filter.

### CODE REVIEW

**Comment by nhatnm0612** [comment]
@jeremi I really need the fields list that is not allow to search in custom filter

**Comment by jeremi** [comment]
@nhatnm0612 any field that we are not using in the UI can be excluded for now.

File: spp_custom_filter/models/custom_filter_mixin.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        Args:
+            :allfields: list of fields to document, all if empty or not provided
+
+            :attributes: list of description attributes to return for each field,
+                all if empty or not provided
+
+        Returns:
+            type: dictionary
+        """
```
Comment (jeremi) [inline spp_custom_filter/models/custom_filter_mixin.py]: Use restructured format

**Review by jeremi — CHANGES_REQUESTED** [review]

File: spp_custom_filter/models/custom_filter_mixin.py (hunk 1 above)
Comment (nhatnm0612) [reply spp_custom_filter/models/custom_filter_mixin.py]: I done it, please re-review

**Review by jeremi — APPROVED** [review]

**Pull request merged**